systems/plants/RigidBodyFlatTerrain.m

```diff
@@ -3,13 +3,18 @@
 %  everywhere
   
   methods 
-    function obj = RigidBodyFlatTerrain()
+    function obj = RigidBodyFlatTerrain(varargin)
+      % Construct a flat terrain map at a given z height (default 0)
+      p = inputParser();
```

Comment:
actually, by default, you can pass an options struct into an input parser and it will just work. Check out the StructExpand option here: http://www.mathworks.com/help/matlab/ref/inputparser-class.html

So, you can do things like:

```
p.parse(varargin{:})
call_my_subfunction(p.Results) %which just passes the same options struct to the subfunction
```

You can even do more clever things like:

```
p.KeepUnmatched = true;
p.parse(varargin{:});
call_my_subfunction(p.Unmatched); % this calls my subfunction, passing in only the arguments that were *not* consumed by the current input parser
```